tools/import-verifier/import-verifier.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+const (
+	rootPackage = "github.com/openshift/origin"
+)
+
+// Package is a subset of cmd/go.Package
+type Package struct {
+	ImportPath   string   `json:",omitempty"` // import path of package in dir
+	Imports      []string `json:",omitempty"` // import paths used by this package
+	TestImports  []string `json:",omitempty"` // imports from TestGoFiles
+	XTestImports []string `json:",omitempty"` // imports from XTestGoFiles
+}
+
+type ImportRestriction struct {
+	// BaseImportPath is the root of the package tree
+	// that is restricted by this configuration
+	BaseImportPath string `json:"baseImportPath"`
```

Comment:
we need exceptions for that path very soon. E.g.:

everything under `staging/src/k8s.io/kube-gen`, but do not apply the restrcitions `staging/src/k8s.io/kube-gen/test/...`.

Compare https://github.com/kubernetes/kubernetes/pull/49501/commits/3d94b5fe160218c146467c5cbf06ee9850004099